### ISSUE TITLE
Add Example codeblock to the docstrings of important functions

### ISSUE THREAD

**Issue by jwmueller** [cleanlab/cleanlab#366]
Especially the methods in the `cleanlab.dataset` module could benefit from their docstrings containing a simple codeblock example of how to use the method.  If adding codeblock examples, make sure to build the docs locally to verify your additions look good, before you open a PR.  Please include before/after screenshots in your PR description as well!

**Comment by jwmueller** [comment]
Once we have a few example codeblock in the docstrings, it would be nice to add testing capabilities in CI that automatically run these example codeblocks as additional tests.

**Comment by Steven-Yiran** [comment]
I am working on this issue.

**Comment by Steven-Yiran** [comment]
Hi @jwmueller, I am wondering if I can get some feedback on this code example I added for the cleanlab.dataset.find_overlapping_classes method. Right now I assumed the data and labels are provided, so the code example will not be runnable. I can work on more methods once I know a good standard of codeblock in the docstrings. 

Here are the before/after screenshots:
Before:
<img width="815" alt="before" src="https://user-images.githubusercontent.com/60683228/230175673-d02e20f8-d7dd-428e-b9e6-48f0fd8c1568.png">
After:
<img width="729" alt="after" src="https://user-images.githubusercontent.com/60683228/230175681-11efea1f-308c-45b0-8946-2bf0b4263111.png">

**Comment by jwmueller** [comment]
Looking good! Can you open a PR with these changes so we can move the discussion there?

In general, this style of docstring is looking good and we'll want to later upgrade the example to actually be runnable (eg. by defining the `get_data_labels(yourFavoriteDataset)` method to load the necessary objects behind the scenes.

Re your specific choice of example, we generally prefer to show the most general example that uses the earliest possible arguments of the method rather than later optional args.  In this case, the canonical args for this `find_overlapping_classes` method would be: `labels, `pred_probs` rather than `confident_joint`. 

Does using `pred_probs` instead make sense?  You can get them from `yourFavoriteModel` without the `CleanLearning` wrapper via: sklearn's [cross_val_predict](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_predict.html), as demonstrated in this tutorial:  https://docs.cleanlab.ai/stable/tutorials/tabular.html

**Comment by Steven-Yiran** [comment]
Yes, using the `pred_probs` makes sense. Thanks a lot for the clarification! I will go ahead and open a PR.

**Comment by Steven-Yiran** [comment]
Hi @jwmueller, I am trying to add doctest code for the examples in the docstrings. However, there is a formatting problem. Specifically, in order for doctest to interpret a docstring code as a valid python expression, I have to add backslash (\\) to every new line like this:
<img width="300" alt="Screen Shot 2023-04-14 at 22 53 46" src="https://user-images.githubusercontent.com/60683228/232179742-07392979-0e0b-4fe5-bea7-da6e7aab9397.png">
But this will result in bad formatting in the docstring examples (i.e. the cross_val_predict() call):
<img width="300" alt="Screen Shot 2023-04-14 at 22 54 07" src="https://user-images.githubusercontent.com/60683228/232179756-b1d49adb-ec7c-460b-acd6-ed230b83344c.png">

I am not sure if I should change the function call to just one line. Do you have any suggestions? Thanks!

**Comment by jwmueller** [comment]
Good question! Yes how about just changing the function call to be just one line.

**Comment by Steven-Yiran** [comment]
I have changed the style of the examples a bit so that with the help of some doctest setup code, this example is actually runnable with `make doctest`. 

<img width="600" alt="health_summary" src="https://user-images.githubusercontent.com/60683228/232255857-9ebb92c6-239b-4f73-a01b-975754615c49.png">


Is this a good way to write the examples? Also, I am not sure how to add testing abilities in CI to automatically run the doctests.

**Comment by jwmueller** [comment]
Can you open a PR with the fixed examples that are runnable via `make doctest`? Our team can then help figure out how to make this run inside our CI in your PR.

This style of example you showed here looks good to me, but I'd delete the final line that only has a comment, since the documentation already explains what the returned object is.

**Comment by ubaidahmad786** [comment]
Hi @jwmueller! I'm a beginner looking to make my first contribution to Cleanlab. I'd love to work on adding example codeblocks to docstrings in cleanlab.dataset. Could you please assign this issue to me?
